.github/workflows/test-apt-packages.yml

```diff
@@ -0,0 +1,82 @@
+  workflow_call:
+    inputs:
+      pkg_type:
+        required: true
+        type: string
+      revision_name:
+        required: true
+        type: string
+
+jobs:
+  TestPackagesWith:
+    strategy:
+      matrix:
+        distro:
+          - "debian:stretch"
+          - "debian:buster"
+          - "debian:bullseye"
+          - "debian:testing"
+          - "debian:sid"
+          - "ubuntu:xenial"
+          - "ubuntu:bionic"
+          - "ubuntu:focal"
+          - "ubuntu:impish"
+      fail-fast: false
+    runs-on: ubuntu-latest
+    container: ${{ matrix.distro }}
+    steps:
+      - name: Install prerequisites for adding 3rd party repository
+        run: |
+          apt-get update -qq
+          apt-get install --yes wget gnupg2 ca-certificates apt-transport-https
+
+      - name: Add nightly OSE repository
+        if: inputs.pkg_type == 'nightly'
+        run: |
+          wget -qO - https://ose-repo.syslog-ng.com/apt/syslog-ng-ose-pub.asc | gpg --dearmor > /usr/share/keyrings/ose-repo-archive-keyring.gpg
+          echo "deb [trusted=yes] https://ose-repo.syslog-ng.com/apt/ ${{ inputs.pkg_type }} $(echo ${{ matrix.distro }} | sed 's/:/-/g')" | tee --append /etc/apt/sources.list.d/syslog-ng-ose.list
+
+      - name: Add stable OSE repository
+        if: inputs.pkg_type == 'stable'
+        run: |
+          wget -qO - https://ose-repo.syslog-ng.com/apt/syslog-ng-ose-pub.asc | gpg --dearmor > /usr/share/keyrings/ose-repo-archive-keyring.gpg
+          echo "deb [signed-by=/usr/share/keyrings/ose-repo-archive-keyring.gpg] https://ose-repo.syslog-ng.com/apt/ ${{ inputs.pkg_type }} $(echo ${{ matrix.distro }} | sed 's/:/-/g')" | tee --append /etc/apt/sources.list.d/syslog-ng-ose.list
+
+      - name: Install the last but one syslog-ng OSE package
+        run: |
+          apt-get update -qq
+          DEBIAN_FRONTEND=noninteractive apt-get install --yes syslog-ng=$(apt-cache madison syslog-ng | sed -n 2p | awk -F"|" '{print $2}' | sed 's/ //g')
+
+      - name: Upgrade to the latest syslog-ng OSE package
+        run: |
+          DEBIAN_FRONTEND=noninteractive apt-get install --yes syslog-ng
+
+      - name: Get syslog-ng revision
+        run: |
+          syslog-ng -V
+          echo "::set-output name=REVISION::$(syslog-ng -V | grep Revision | cut -d" " -f2)"
+        id: syslog_ng_revision
+
+      - name: Check if revision is snapshot revision
+        if: inputs.revision_name == 'snapshot'
+        run: |
+          echo ${{ steps.syslog_ng_revision.outputs.REVISION }} | grep ${{ inputs.revision_name }}
+
+      - name: Check if revision is stable revision
+        if: inputs.revision_name == 'stable'
+        run: |
+          echo ${{ steps.syslog_ng_revision.outputs.REVISION }} | egrep "^[0-9]{1}.[0-9]{1,2}.[0-9]{1}-[0-9]{1}$"
+
+      - name: Check if installed package version matches with install revision
+        run: |
+          echo "Installed revision value: ${{ steps.syslog_ng_revision.outputs.REVISION }}"
```

Comment:
Can we add a `dpkg -l | grep syslog-ng` command too for debugging purposes here, please?

The actual checking with `dpkg -l` doesn't show the list of installed packages if the validation fails:
https://github.com/mitzkia/syslog-ng/runs/5095931848?check_suite_focus=true
(Thumbs up that you tried out failing cases of version validation)

Reply:
Thanks for the note, agreed with it. Fixed.